autokeras/auto_model.py

```diff
@@ -77,6 +78,10 @@ class AutoModel(object):
             The input node(s) of the AutoModel.
         outputs: A list of Node or Head instances.
             The output node(s) or head(s) of the AutoModel.
+        pipelines: An instance or a list of Pipeline corresponding to each of the
+            input node, which transforms tensorflow.data.Dataset instance using
+            tensorflow.data operations before feeding into the neural network.
+            Defaults to None.
```

Comment:
"""
pipelines: An instance or list of `Pipeline` objects corresponding to each
            AutoModel input, to preprocess a `tf.data.Dataset` before passing it to the model.
            Defaults to None (no external preprocessing).
"""

In addition, I'm not sure "pipeline" really conveys what this does. The defining characteristic of this API is that it defines a set of transformations that are applied to the input data *before* hitting the model. So "preprocessor" would work better (other options are possible).